airbyte-webapp/src/views/Connector/ServiceForm/components/Sections/VariableInputFieldForm.tsx

```diff
@@ -0,0 +1,82 @@
+import { useField } from "formik";
+import { FormattedMessage } from "react-intl";
+import { useAsync, useEffectOnce } from "react-use";
+import * as yup from "yup";
+
+import { Button, ModalBody, ModalFooter } from "components";
+
+import { FormGroupItem, FormObjectArrayItem } from "core/form/types";
+
+import { useServiceForm } from "../../serviceFormContext";
+import { FormSection } from "./FormSection";
+interface VariableInputFormProps {
+  formField: FormObjectArrayItem;
+  path: string;
+  item?: unknown;
+  disabled?: boolean;
+  onDone: (value: unknown) => void;
+  onCancel: () => void;
+}
+
+export const VariableInputFieldForm: React.FC<VariableInputFormProps> = ({
+  formField,
+  path,
+  item,
+  disabled,
+  onDone,
+  onCancel,
+}) => {
+  const hiddenPath = `__hidden_${path}`;
```

Comment:
To my understanding of how the dialog is designed at the moment, this is part of the overall ServiceForm and we just use a special `__hidden_${path}` variable to attach the fields under within the form. Could it make sense to potentially have this modal opening it's own Formik form (and being detached from the ServiceForm), and just communicate (as it already does) it's values onDone towards the ServiceForm which will store it?

This might also allow us to do better validation by just using the subpart of the validationSchema as a new validationSchema for the modal's Formik form?